### PR TITLE
Add tenant UUID to getKey

### PR DESCRIPTION
**Pull request by ntarora**
# Add tenant uuid (optional) to get key, to enable per tenant uaa.

- add the option to pass tenant uuid to getKey. This is done to accommodate per tenant UAA when the UAA instance url is unknown and only known by the proxy'd trusted issuer url.

- Optional should not break any functionality

- Test added + formatting

### CODE REVIEW

**Comment by bsyk** [comment]
Thanks for the contribution.
Please add a test that uses the tennantUUID and ensure the requests and cache entry are treated correctly.
Generally when making PRs, try to separate logic and formatting changes, so that it's easier to see what's been changed.  2 clear PRs are better than 1 big one :)

File: index.js [hunk 1]
```diff
@@ -13,83 +13,99 @@ token_utils._tokenCache = tokenCache; // Exposed for testing
...
-                debug('Error reading token key from', keyURL, err);
-                reject(err);
-            });
-        }
-    });
+const getKey = (keyURL, tenantUuid) => {
+  // URL for the token is <UAA_Server>/token_key
+  return new Promise((resolve, reject) => {
+    // Check the cache
+    if (tenantUuid && oauthKeyCache[keyURL + tenantUuid]) {
```
Comment (bsyk) [inline index.js]: You may want to consider a delimiter between the URL and UUID, in case it's accidentally passed as an empty string.

Comment (ntarora) [reply index.js]: Good catch. Will add one.

**Comment by ntarora** [comment]
You are right will remove the formatting changes and focus on the core changes for tenant uuid.

The two test cases for tenant uuid and caching were added, but probably hard to catch with all the changes.

**Pull request closed without merging**